src/Features/Core/Portable/Completion/CompletionItemRules.cs

```diff
@@ -6,6 +6,28 @@
 
 namespace Microsoft.CodeAnalysis.Completion
 {
+    public enum CompletionItemSelectionBehavior
+    {
+        Default,
+
+        /// <summary>
+        /// If no text has been typed, the item should be soft selected. This is appropriate for 
+        /// completion providers that want to provide suggestions that shouldn't interfere with 
+        /// typing.  For example a provider that comes up on space might offer items that are soft
+        /// selected so that an additional space (or other puntuation character) will not then 
+        /// commit that item.
+        /// </summary>
+        SoftSelection,
```

Comment:
I'm a little surprised I didn't see something for "new" preselection in C# here.  I would expect to be able to type: `Foo f = new (` and get `Foo f = new Foo(`.